draft-ietf-httpbis-semantics-latest.xml

```diff
@@ -732,7 +732,7 @@ Content-Type: text/plain
    The target of an HTTP request is called a "<x:dfn>resource</x:dfn>". 
    HTTP does not limit the nature of a resource; it merely
    defines an interface that might be used to interact with resources.
-   Each resource is identified by a Uniform Resource Identifier (URI), as
+   Most resources are identified by a Uniform Resource Identifier (URI), as
    described in <xref target="uri"/>.
```

Comment:
Framing it this way suggests that the server itself (`asterisk-form`) and other network endpoints (`authority-form`) are resources without URIs.  This isn't necessarily wrong, but feels like an odd formulation.  Would it be better to say that these oddball methods can address things other than resources?

Reply:
Lots of resources don't have URIs; e.g., the state in the browser is a resource, but it usually doesn't have one.

Reply:
Well, now, that depends on whose spec is talking.

The word resource implies you can use it at some later time, hence re-source it. That is difficult without an identifier. In HTTP, "resource" is a defined term and every HTTP resource has at least one (usually much more than one) URI, not limited to http(s) URIs, but a message recipient might not know any of those URIs. The server itself and other network endpoints have URIs, but we don't use those URIs in the method-specific HTTP/1.1 requests for historical (foolish) reasons. Hence, the target of an HTTP request (semantics) is a resource, but the request-target (bits on wire) may only transmit a small portion of that resource's identifying data obtained from a URI.

The word resource is also used by WHATWG specs, with an incompatible definition, due to NIH syndrome. Whether or not the state within a browser is a resource, or has a URI (which is almost impossible to avoid within a browser given about: URIs), is another very long and painful story.

Reply:
Looking through the OED, that's a very creative interpretation of the etymology of _resource_, but nevertheless interesting (and that was the meaning I was thinking of, not the WHATWG's).

Regardless, could you review the PR?